subsurface/geometry/mesh.py

```diff
@@ -0,0 +1,398 @@
+"""These are classes that are point based and managed via Pandas DataFrames.
+
+All data is tracked within internal DataFrames that we validate.
+
+This is for holding general "mesh"-like data structures: point clouds,
+linesets, triangulated surfaces, tetrahedralized volumes, octree grids, etc.
+
+Regularly gridded dataset will NOT be managed by these classes but will use
+``xarray`` under the hood.
+
+"""
+
+import numpy as np
+import pandas as pd
+
+
+from .common import Common
+from .errors import PyVistaImportError
+
+
+class PointSet(Common):
+    """Base class for pointset based data structures.
+
+    This class contains a data frame of vertices (points) and a data frame
+    of data associated with each of those vertices.
+
+    """
+
+    def __init__(self,
+                 points=None,
+                 point_column_names=('x', 'y', 'z'),
+                 ):
+        """
+        Initialize the pointset from a datafame.
+
+        Parameters
+        ----------
+        points : pd.DataFrame
+            A dataframe that has XYZ coordinates that are named as such.
+            Additional columns will be tracked as point data.
+
+        """
+        self._df_points = pd.DataFrame(columns=['X', 'Y', 'Z'])
+        self._df_point_data = pd.DataFrame()
```

Comment:
is it worth it to have two dataframes or should we just make one and we just make views:

self._df = pd.Data...
self._df_points is self._df['X', 'Y', 'Z']
self._df_point data is self._df[the rest]